doc/developer/design/20230814_unbilled_replicas.md

```diff
@@ -0,0 +1,203 @@
+# Unbilled replicas
+
+- [#20317](https://github.com/MaterializeInc/materialize/issues/20317)
+
+## Context
+
+Support requires the ability to dynamically add resources to a user's environment
+without incurring substantial costs for the user. For example, when support
+notices that after a version upgrade, Materialize needs additional compute
+resources, support should be able to add such resources for a limited amount of
+time without requiring explicit approval and complicated reimbursement for the
+customer.
+
+We extend Materialize with a concept of unbilled replicas, which server compute
+traffic but are not charged to a customer's account.
+
+To achieve this, we extend clusters and their replicas with concept of replica sets.
+A replica set defines properties shared by a partition of replicas within a cluster.
+We allow internal users to create replica sets with specific billing information to add free or cost-reduced resouces to user environemnts, without changing the structure of a user's deplyoment.
+
+## Goals
+
+We introduce unbilled clusters by the following means:
+* Introduce replica sets.
+* Add a billing property to replica sets.
+* Record replica set properties in audit events.
+
+## Non-Goals
+
+We do not plan to do the following:
+* Allow customers general access to replica sets.
+
+## Overview
+
+We introduce the concept of replica sets, which capture partitions
+of replicas sharing configuration aspects. Replica sets can co-exist, and a default
+replica set captures the current configuration of a cluster. Each replica is a member of a single
+replica set.
+
+At this time, we only support replica sets for internal users, but do not
+give customers the permission to directly interact with replica sets. This
+is to give us time to validate the feature, and decide on reserved names.
+
+## Detailed description
+
+We extend clusters to support multiple replica configurations, and name each a replica set.
+A replica set describes the properties of a partition of replicas within a cluster.
+A cluster can have several replica sets, and each replica is part of a single replica set.
+
+Multiple replica set within a cluster can co-exist simultaneously, and can have different parameters.
+We do not change that all replicas in a cluster execute equivalent commands and participate equally in producing responses.
+
+```sql
+-- Creating a managed replica set without billing
+CREATE REPLICA SET cluster.support WITH (SIZE '3xlarge', BILLED AS 'unbilled');
```

Comment:
Drop the `WITH` for parity with `CREATE CONNECTION`, which has the best syntax. The `WITH` is only necessary in e.g. `CREATE SOURCE` where there are non-`WITH` options. 

This is a bit at variance with `CREATE CLUSTER`'s syntax, which does not require the parens, but I've been meaning to update the `CREATE CLUSTER` syntax to _allow_ the enclosing parens. This would be backwards compatible and we could adjust the docs to use only the consistent syntax.